src/hooks/useMap.ts

```diff
@@ -6,17 +6,20 @@ export default function useMap<T>(defaultValue: T[], keyMapper: (value: T) => an
   const add = (value: T) => {
     map.set(keyMapper(value), value);
     setMap(new Map(map));
+
+    return map;
```

Comment:
This should return the new map but is returning the old, mutated map.

Reply:
That's a good point, I will modify it to return the new map.